### ISSUE TITLE
ensure_sequence_of_iterables : ValueError for empty list. 

### ISSUE THREAD

**Issue by VolkerH** [napari/napari#3079]
## 🐛 Bug

While chasing down some problems with displaying multi-channel, pyramidal ome.zarr files through a plugin
I noticed a problem in `ensure_sequence_of_iterables`:

https://github.com/napari/napari/blob/3bd58bde6858eb4d9f916fe32df262a4ebe1a24f/napari/utils/misc.py#L151

When the `obj` passed in is an empty list `[]`, an index error is raised:

```
~/Github/napari/napari/utils/misc.py in ensure_sequence_of_iterables(obj=[], length=4)
    182     """
    183     print(f"in ensure sequence of iterables {obj}")
--> 184     if obj is not None and is_sequence(obj) and is_iterable(obj[0]):
        obj = []
        global is_sequence = <function is_sequence at 0x7f45f5aa8550>
        global is_iterable = <function is_iterable at 0x7f45f5aa84c0>
    185         if length is not None and len(obj) != length:
    186             raise ValueError(

IndexError: list index out of range
```

In the ideal world (no upstream errors) there probably shouldn't be empty lists passed into this function. 
However, it is not quite clear to me what the desired failure mode is for (possibly incorrect) input such as
`[]` or `None`. For `[]` one gets the above `IndexError`, for `None` one will get an iterator that will repeatedly 
output `None`.
I am not sure whether the latter (repeated `None` ) is desired behaviour or an error should be raised.

**Comment by tlambert03** [comment]
looking only at this function, i think I agree that an empty list qualifies as a sequence of iterables (albeit an "empty" sequence of iterables).  I think the only place this function is used in in `stack_utils:split_channels`  (where, to my reading, an empty list should be fine as well ... or at least should be handled there instead).

I think the behavior for an empty list should be to return an empty list (unless `length` is provided and not 0, in which case it should be a valueError

**Comment by sofroniewn** [comment]
Just noting I think this will be closed by #3140 - have we added a test for it over there though @brisvag @alisterburt?

**Comment by brisvag** [comment]
Yes, there a couple more tests and it should also close automatically (due to the commit message being "fix #3079") :)
